conans/client/build/autotools_environment.py

```diff
@@ -339,6 +341,10 @@ def append(*args):
         tmp_compilation_flags = copy.copy(self.flags)
         if self.fpic:
             tmp_compilation_flags.append(pic_flag(self._conanfile.settings))
+        if tools.is_apple_os(self._os) and self._os_version:
+            flag = tools.apple_deployment_target_flag(self._os,
+                                                      self._os_version)
+            tmp_compilation_flags.append(flag)
```

Comment:
this flag needs to be added carefully
we know from the https://github.com/conan-community/community/issues/304, some users may specify explicit deployment target, e.g. in CFLAGS/CXXFLAGS. we shouldn't overwrite that, otherwise it may break some use cases (e.g. Catalist)

Reply:
Ok thank you for your feedback! What do you think would be the best option?

1. Close this PR and apply the flag per recipe (like I did in #3170
2. Apply the same checks done here: https://github.com/conan-io/conan-center-index/pull/585/files
3. Move the flag addition to another part of the code more safe or logical
4. 2 and 3

Reply:
№2 looks as the best option for me. per recipe doesn't scale well.

Reply:
I found a way to test by simply setting an environment variable: https://github.com/conan-io/conan/pull/7862/files#diff-809c0e7ca852b9746fa8614ed3a57074892e114a74c0072cfc8d92fbad78e195R691-R695